enterprise/backend/src/metabase_enterprise/semantic_search/indexer.clj

```diff
@@ -89,6 +91,14 @@
       (log/debugf "Marked indexer as stalled for index %s at %s" (:table-name index) timestamp))
     timestamp))
 
+(defn- observe-poll-to-poll-interval [poll-result watermark]
+  (let [{new-poll-time :poll-time} poll-result
+        {:keys [last-poll-time]} watermark]
+    (when last-poll-time
+      ;; distance between polls
+      (let [idle-time-ms (- (inst-ms new-poll-time) (inst-ms last-poll-time))]
```

Comment:
this is not necessarily idle time (and may want to change this), idle interval you would want to subtract the time spent working right?

That said I expect this interval is still useful to determine (along with other measurements) how often we loop back to poll, this lets us tune variables that would depend on poll frequency. The percentile distribution lets us know how often we are fully idle (yielded), in the tighter loops. There may be a better way to measure this.

Reply:
Overall idle vs busy can still be known with these counters AFAICT

```
   (prometheus/counter :metabase-search/semantic-indexer-loop-ms
       {:description "Total number of ms spent in the semantic search indexer loop"})
   (prometheus/counter :metabase-search/semantic-indexer-sleep-ms
       {:description "Total number of ms the semantic indexer loop had control but was asleep"})
```